src/ray/rpc/grpc_client.h

```diff
@@ -109,25 +109,6 @@ class GrpcClient {
     stub_ = GrpcService::NewStub(channel_);
   }
 
-  GrpcClient(const std::string &address,
-             const int port,
-             ClientCallManager &call_manager,
-             int num_threads,
-             bool use_tls = false)
-      : client_call_manager_(call_manager), use_tls_(use_tls) {
-    grpc::ChannelArguments argument = CreateDefaultChannelArguments();
-    grpc::ResourceQuota quota;
-    quota.SetMaxThreads(num_threads);
-    argument.SetResourceQuota(quota);
-    argument.SetInt(GRPC_ARG_ENABLE_HTTP_PROXY,
-                    ::RayConfig::instance().grpc_enable_http_proxy() ? 1 : 0);
-    argument.SetMaxSendMessageSize(::RayConfig::instance().max_grpc_message_size());
-    argument.SetMaxReceiveMessageSize(::RayConfig::instance().max_grpc_message_size());
-
-    channel_ = BuildChannel(address, port, argument);
-    stub_ = GrpcService::NewStub(channel_);
-  }
```

Comment:
change places where this constructor is used too? Are the other uses casting num_threads to bool rn?

Reply:
I don't find it used anywhere via simple `grep`, but I will wait for CI results.

Reply:
https://github.com/ray-project/ray/blob/2385b189f08b0306d544d773607190029c65ac75/src/ray/rpc/object_manager/object_manager_client.h#L51-L52 this seems to be the constructor used here, i think now it's just using one above this one instead and implicitly coverts the int to a bool

Reply:
Good catch! Updated.

Reply:
How do we make sure we removed all usages of this constructor and didn't accidentally call the one above by converting int to bool?

Reply:
I made an extra test PR: https://github.com/ray-project/ray/pull/49283, which throws exception on the grpc client overload; and it turns out all CI tests pass.